### ISSUE TITLE
Update Git for Windows version to account for recent GitHub breakage

### ISSUE THREAD

**Issue by plioi** [gitextensions/gitextensions#4523]
**Do you want to request a *feature* or report a *bug*?**

An upgrade to the Git for Windows version included at install time.

**What is the current behavior?**

https://github.com/Microsoft/Git-Credential-Manager-for-Windows#notice-experiencing-github-pushfetch-problems GitHub recently disabled weak encryption support. The Git Credential Manager for Windows page describes the problem and advises people update their Git for Windows version as a fix (v2.16.0).

The problematic behavior is described on the linked page above.

**Environment you encounter the issue:**

- GitExtensions version: 2.50.02
- GIT version: 2.10.1.windows.1
- OS version: Windows 10

**Did this work in previous version of GitExtensions (which)?**

Yes, all.

**Comment by gerhardol** [comment]
I propose that git is removed from the installer and a download link is provided instead. #4515

**Comment by plioi** [comment]
+1 to gerhardol. In the past, I would install git extensions not only for the great history viewer, but also because it just made installing on windows so much easier. If that's no longer relevant, separating the two seems like a good idea.

FWIW, upgrading git didn't help my root problem either, but that's not GitExtensions' concern.

**Comment by RussKie** [comment]
Try upgrading putty.
IIRC in another thread people said it helped

**Comment by gerhardol** [comment]
PuTTY must be updated for ssh, Git Credential manager for https

Proposed way forward, both 2.51 and 3.00:

1. GE documents in the wiki the recommended Git version for each GE version including issues

2. Stop bundling Git (and KDIff3) with GE. There still need to be a GitExtensions-*-SetupComplete.msi to be downloaded I assume.

3. Provide a download link to Git for Windows if no Git installed (for Mono, no link but a message).

4. Warn if the Git version seen when starting up is too old At every startup), info (suppressible) if older than recommended.

**Comment by pmiossec** [comment]
I don't know if, due to the changes that github have made deprecating some security protocols, we should not release soon a last version with git and putty upgraded (to minimise the fact that new users download the old setup that won't work) and then only after remove git from the setup...

**Comment by gerhardol** [comment]
PuTTY 0.70 is in master and release/2.51
Git 2.16.1.2 is in master. Should be updated in 2.16.2.1 and cherry-picked to release/2.51

But the proposed changes are not so difficult and a better solution in my opinion.
Good to hear other opinions before we decide what we want to do, the to decide what should be done in the short run

**Comment by plioi** [comment]
FWIW, I got around my problem after realizing I was running an older git living under Program Files (x86) while all the newer installs were putting things under Program Files. It'd be nice if installing git started by detecting that legacy install, but that's probably more Git for Windows' problem and not Git Extensions' responsibility at all.

**Comment by BinarysoftInc** [comment]
Make sure you are using the correct path for the Putty.

![putty](https://user-images.githubusercontent.com/4038731/36778673-73808074-1c93-11e8-86d9-5c172db9414c.png)

**Comment by patricktokeeffe** [comment]
@BinarysoftInc The default path I see for PuTTY is `C:\Program Files (x86)\GitExtensions\PuTTY\`

**Comment by patricktokeeffe** [comment]
fwiw my temporary workaround is to launch Git Bash and use `git clone`

**Comment by RussKie** [comment]
Yesterday on one my computers I had exactly the problem described.
I've updated git and putty and everything is working again.
